appcelerator/titanium_mobile
# fix(android): HTTPClient "responseData" blob returns 0 width/height for images over 512kb

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-27108

**Summary:**
Issue happened because our Java `TiHTTPClient` class creates a temp file for responses that are bigger than 512 KB, but we create a "responseData" blob which wraps that empty temp file at the start of the download. The blob's constructor will fail to load image info since the temp file is currently empty. Solution is to call the blob's `loadBitmapInfo()` method once the download finishes.

**Test:**
1. Acquire an Android device that has Internet access.
2. Build and run the below code on that Android device.
3. Tap on the "Download" button.
4. Verify alert dialog message shows: "Image Size: 1280x1024"  _(Should not be 0x0)._
5. Verify an image appears in the background.

```javascript
var imageUrl = "https://jira.appcelerator.org/secure/attachment/66701/NasaMars.png";

var activityIndicator = Ti.UI.createActivityIndicator({
	indicatorColor: (Ti.App.iOS ? "blue" : "white"),
	style: Ti.UI.ActivityIndicatorStyle.BIG,
});

var window = Ti.UI.createWindow({
	title: "Download Image Test",
	exitOnClose: false,
});
var imageView = Ti.UI.createImageView({
	autorotate: true,
	top: 0,
});
window.add(imageView);
var downloadButton = Ti.UI.createButton({
	title: "Download",
	bottom: "20dp",
});
downloadButton.addEventListener("click", function(e) {
	var httpClient = Ti.Network.createHTTPClient({
		onload: function(e) {
			activityIndicator.hide();
			downloadButton.enabled = true;
			var imageBlob = httpClient.responseData;
			imageView.image = imageBlob;
			Ti.API.info("@@@ onload() imageBlobSize: " + imageBlob.width + "x" + imageBlob.height);
			alert("Image Size: " + imageBlob.width + "x" + imageBlob.height);
		},
		onerror: function(e) {
			Ti.API.info("@@@ onerror()");
			alert("Download Failed");
			activityIndicator.hide();
			downloadButton.enabled = true;
		},
	});
	activityIndicator.show();
	imageView.image = null;
	downloadButton.enabled = false;
	httpClient.open("GET", imageUrl);
	httpClient.setRequestHeader("Accept-Encoding", "identity");
	httpClient.send();
});
window.add(downloadButton);
window.add(activityIndicator);
window.open();
```
